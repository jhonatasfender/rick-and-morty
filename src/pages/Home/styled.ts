import styled, { css } from 'styled-components'

export const Image = styled.img.attrs({ src: '/img/first.jpg' })`
  height: 100vh;
  width: 100%;
`

export const BackgroundRed = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.backgroundImage};
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
  `}
`

export const WrapperImage = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
`

export const Title = styled.span`
  ${({ theme }) => css`
    font-size: max(9vw, 3rem);
    font-weight: bold;
    position: absolute;
    z-index: ${theme.layers.base};
    color: ${theme.colors.white};
    user-select: none;
  `}
`
