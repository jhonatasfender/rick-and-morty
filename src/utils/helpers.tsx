import { ReactNode } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { render, RenderResult } from '@testing-library/react'

import theme from '../styles/theme'

const queryClient = new QueryClient()

export const renderWithThemeAndRoute = (children: ReactNode): RenderResult =>
  render(
    <QueryClientProvider client={queryClient}>
      <MemoryRouter>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </MemoryRouter>
    </QueryClientProvider>,
  )
