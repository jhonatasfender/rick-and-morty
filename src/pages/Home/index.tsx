import * as S from './styled'

export default function Home(): JSX.Element {
  return (
    <S.WrapperImage>
      <S.Title>Rick and Morty</S.Title>
      <S.BackgroundRed />
      <S.Image />
    </S.WrapperImage>
  )
}
