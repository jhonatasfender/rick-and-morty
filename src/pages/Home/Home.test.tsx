import { renderWithThemeAndRoute } from '../../utils/helpers'
import Home from './'

describe('<Home />', () => {
  it('should load the component with default values', async () => {
    const { container } = renderWithThemeAndRoute(<Home />)

    expect(container).toMatchSnapshot()
  })
})
