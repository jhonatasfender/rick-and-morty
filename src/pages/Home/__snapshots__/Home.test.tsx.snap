// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should load the component with default values 1`] = `
.c3 {
  height: 100vh;
  width: 100%;
}

.c2 {
  background-color: rgb(255 0 0 / 37%);
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: max(9vw,3rem);
  font-weight: bold;
  position: absolute;
  z-index: 10;
  color: #ffffff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      Rick and Morty
    </span>
    <div
      class="c2"
    />
    <img
      class="c3"
      src="/img/first.jpg"
    />
  </div>
</div>
`;
